### DISCUSSION TITLE
How can I cancel a `StreamSubscription` in my Notifier class?

### DISCUSSION THREAD

**Discussion by Biplab-Dutta** [rrousselGit/riverpod#3752] [Q&A]
I have a class that extends `AutoDisposeNotifier` and I have a `NotifierProvider.autoDispose` as well. How can I cancel the stream subscription when the notifier is disposed. Or when the page is disposed. I can't do `ref.onDispose` in my NotifierProvider as it doesn't provide ref object in the callback. Is there any other recommended approach?

**Comment by rrousselGit** [comment]
Notifiers have a ref property, so you can use ref.onDispose

**Reply by Biplab-Dutta** [reply]
I don't get it. Am I supposed to create a method in my class (that extends Notifier) with `ref.onDispose` inside the method and have that method called in my State.dispose? I can't do `ref.onDispose` in the `NotifierProvider` as I don't have a ref in the callback. 
Also, using ref inside State.dispose isn't recommended. So, I am clearly wrong here.
@rrousselGit

**Reply by Biplab-Dutta** [reply]
I did it inside of the build method in my Notifier. I hope I am doing it right.
